query/encode.go

```diff
@@ -166,6 +173,8 @@ func reflectValue(values url.Values, val reflect.Value) error {
 				del = ','
 			} else if opts.Contains("space") {
 				del = ' '
+			} else if opts.Contains("key") {
```

Comment:
I probably went for `key` rather than `brackets` because `comma` and `space` refer to what the delimiter character is to serliaize the value into a single query parameter. However, what `key` does is adding several query parameters with the same key.

With that being said, I agree key might still be too obscure. Here are some alternative ideas (some crazier than others):
 - `repeatkey`, `multikey` or `multivalue`
 - `splat` or `variadic` since this filed will be split in multiple url.Value
 - `...` in the spirit of Go's `...` operator

I would discard `[]` or `brackets` since it might be too aligned with `comma` and `space` which refer to delimiter characters.